server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -785,8 +785,12 @@ private void tryToUpdateBatchInsertLastCache(InsertTabletPlan plan, Long latestF
         continue;
       }
       // Update cached last value with high priority
-      ((MeasurementMNode) node.getChild(measurementList[i]))
-          .updateCachedLast(plan.composeLastTimeValuePair(i), true, latestFlushedTime);
+      MeasurementMNode tmpMeasurementNode = null;
+      if (node != null) {
+        tmpMeasurementNode = (MeasurementMNode) node.getChild(measurementList[i]);
+      }
+      IoTDB.metaManager.updateLastCache(node.getFullPath() + IoTDBConstant.PATH_SEPARATOR + measurementList[i],
```

Comment:
The String concat is an overhead in single node IoTDB, could we avoid this?

Reply:
How about the following code?
` if (tmpMeasurementNode != null) {
        IoTDB.metaManager.updateLastCache(tmpMeasurementNode.getFullPath(),
          plan.composeLastTimeValuePair(i), true, latestFlushedTime, tmpMeasurementNode);
      } else {
        IoTDB.metaManager.updateLastCache(node.getFullPath() + IoTDBConstant.PATH_SEPARATOR + measurementList[i],
          plan.composeLastTimeValuePair(i), true, latestFlushedTime, tmpMeasurementNode);
      }`